lib/rspec/core/example.rb

```diff
@@ -565,13 +566,23 @@ def example_skipped?
         # @api private
         # Records the finished status of the example.
         def record_finished(status, finished_at)
-          self.status      = status
-          self.finished_at = finished_at
-          self.run_time    = (finished_at - started_at).to_f
+          self.status = status
+          calculate_run_time(finished_at)
+        end
+
+        # @api private
+        # Populates finished_at and run_time if it has not yet been set
+        def ensure_timing_set(clock)
+          finished_at || calculate_run_time(clock.now)
```

Comment:
I had originally written this as `finished_at or calculate_run_time(clock.now)` in the spirit of http://devblog.avdi.org/2014/08/26/how-to-use-rubys-english-andor-operators-without-going-nuts/ - but rubocop prohibits using `or` so I tweaked it to `||` (which works, but… yeah, what you said).

Anyway, I agree that the `unless` version is more-expressive - thanks for pointing that out.